### ISSUE TITLE
Support request for Linux PowerShell Core

### ISSUE THREAD

**Issue by LaxmiPrasadBotla** [HewlettPackard/POSH-HPOneView#406]
Please fill in as much information as possible to help resolve your issue.

### Expected Behavior
Import module have to happen without errors in Linux. I have this module path listed in PS 
```powershell
$env:PSModulePath.

$ Import-Module HPOneView.410
```
### Actual Behavior
```powershell
Import-Module HPOneView.410
Import-Module : Could not load type 'Microsoft.PowerShell.Commands.OutPrinterCommand' from assembly 'Microsoft.PowerShell.Commands.Utility, Version=6.2.0.0, 

Culture=neutral, PublicKeyToken=31bf3856ad364e35'.
At line:1 char:1
+ Import-Module HPOneView.410
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [Import-Module], TypeLoadException
+ FullyQualifiedErrorId : System.TypeLoadException,Microsoft.PowerShell.Commands.ImportModuleCommand
```
### Steps to reproduce
```powershell
$ Import-Module HPOneView.410 -Verbose
VERBOSE: Loading module from path '/root/XXX/library/HPOneView.410/HPOneView.410.psd1'.
Import-Module : Could not load type 'Microsoft.PowerShell.Commands.OutPrinterCommand' from assembly 'Microsoft.PowerShell.Commands.Utility, Version=6.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.
At line:1 char:1
+ Import-Module HPOneView.410 -Verbose
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [Import-Module], TypeLoadException
+ FullyQualifiedErrorId : System.TypeLoadException,Microsoft.PowerShell.Commands.ImportModuleCommand
```

```PowerShell
# 1. Capture verbose output using the Get-HPOVCommandTrace for HPOneView.310 or newer, or append the -Verbose switch to your Cmdlet call.
# 2. Put your code and/or captured output here.
# 3. Remove this comment block before submitting.
```

### Version Information
HPE OneView PowerShell Library Version (`Get-HPOVVersion` or `$PSLibraryVersion`):
HPE OneView Appliance Version (`Get-HPOVVersion -ApplianceVer`):  
Output from `$PSVersionTable` on your Windows Host:

```PowerShell
# Output from $PSVersionTable
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.2.0
PSEdition                      Core
GitCommitId                    6.2.0
OS                             Linux 3.10.0-957.el7.x86_64 #1 SMP Thu Nov 8 23:39:32 UTC 2018
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by ChrisLynchHPE** [comment]
Unfortunately, none of the libraries support PowerShellCore.  It mainly has to do with certain classes that have not been available in prior DotNetCore that we rely upon.  With the release of PowerShellCore 6.2 and upcoming 7, there are more options for us to look at migrating support to PowerShellCore.  We will continue investigating support for PowerShellCore.

**Comment by ChrisLynchHPE** [comment]
I am revisiting this topic.  I have performed some research on transitioning to .Net Standard 2.0 to support PowerShell 5.1, 6.x and PowerShellCore 6.x.  I will be uploading a new test branch of the 4.20 library that can be used to test with.  However, I will not be looking to release the newer PowerShell library and associated class library until HPE OneView 5.0 is released (which we announced back at HPE Discover Las Vegas 2019).

I'll follow up on this thread when the branch has been committed to the repo here.

**Comment by ChrisLynchHPE** [comment]
Can you also provide the output from $PSversiontable on your Linux system?  Also, what Linux distro are you using?

**Comment by LaxmiPrasadBotla** [comment]
$PSversiontable on my Linux - 

$PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.2.0
PSEdition                      Core
GitCommitId                    6.2.0
OS                             Linux 3.10.0-957.el7.x86_64 #1 SMP Thu Nov 8 23:39:32 UTC 2018
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

And my Linux distro is CenOS.

**Comment by ChrisLynchHPE** [comment]
Thanks.  I have tested the library thus far on Ubuntu 18.04 LTS and Debian 9.  I will be pushing the test branch today.

**Comment by ChrisLynchHPE** [comment]
The [branch](https://github.com/HewlettPackard/POSH-HPOneView/tree/4.20-DotNetStandard-Dev) I mentioned has been created and is open to clone from.  Please read the [README](https://github.com/HewlettPackard/POSH-HPOneView/tree/4.20-DotNetStandard-Dev/README.md) in the branch to find out base requirements, how to use it, known limitations and how to submit feedback.

**Comment by olant** [comment]
Chris,
I cannot find the brach you mention above. Is there any news on supporting PowerShell Core for the 4.20 version?

**Comment by ChrisLynchHPE** [comment]
Unfortunatly, no as it would be a breaking change.  Only the 5.00 library going forward will be built upon .NetStandard 2.x.

**Comment by ChrisLynchHPE** [comment]
This enhancement is addressed in [Release 5.0.2152.1665](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v5.0.2152.1665).

Closing.
